src/entry_writer.rs

```diff
@@ -119,7 +120,7 @@ mod tests {
         let tx = Transaction::new(&mint.keypair(), keypair.pubkey(), 1, mint.last_id());
 
         // NOTE: if Entry grows to larger than a transaction, the code below falls over
-        let threshold = (BLOB_DATA_SIZE / 256) - 1; // 256 is transaction size
+        let threshold = (BLOB_DATA_SIZE / PACKET_DATA_SIZE) - 1; // PACKET_DATA_SIZE is transaction size
```

Comment:
should be this:
let threshold = (BLOB_DATA_SIZE / tx_small_size as usize) - 1;

and no: the NOTE has to stay because the containing "Entry" might have a size that's larger than a small transaction.